source/external-editor.lisp

```
@@ -3,6 +3,44 @@
 
 (in-package :nyxt)
 
+(-> %append-uiop-command ((or string (list-of string)) &rest string) (values (or string (list-of string)) &optional))
+(defun %append-uiop-command (command &rest args)
+  "Appends ARGS to an existing COMMAND (for `uiop:run-program' or
+`uiop:launch-program').
+
+If COMMAND is a string, ARGS is concatenated to it with spaces between the
+arguments.
+
+If COMMAND is a list, ARGS is appended to it.
+
+Signals an error if COMMAND is nil or an empty string."
+  ;; The uiop functions expect either the entire command as a string, or a list
+  ;; of strings with the command as the first element, and each parameter as
+  ;; subsequent elements. Mixing them signals an error. This is the reason for
+  ;; this custom append function.
+  (cond
+    ((null command) (error "Unable to append arguments to a null command."))
+    ((str:emptyp command) (error "Unable to append arguments to an empty command."))
+    ((null args) command)
+    ((stringp command) (uiop:reduce/strcat (list command " " (str:unwords args))))
+    ((consp command) (append command args))))
+
+(export-always '%run-external-editor)
+(defun %run-external-editor (path &optional (program (external-editor-program *browser*)))
+  "Calls `uiop:run-program' with PATH as an extra parameter to PROGRAM.
+PROGRAM defaults to `external-editor-program'"
+  (let ((command (%append-uiop-command program (uiop:native-namestring path))))
+    (log:debug "External editor opens ~s" command)
+    (uiop:run-program command :ignore-error-status t)))
```

Shall we really ignore the status?  What happens on error?

---

I have no idea. The existing calls did this, so I just kept it. I hate swallowing errors though, so it looks scary.

---

Ok, I'll remove this in it's own commit and use your comment as a basis for the commit message.